docs/en/ingest-management/processors/processor-add_cloud_metadata.asciidoc

```diff
@@ -0,0 +1,225 @@
+[[add-cloud-metadata-processor]]
+= Add cloud metadata
+
+++++
+<titleabbrev>add_cloud_metadata</titleabbrev>
+++++
+
+TIP: Inputs that collect logs and metrics use this processor by default, so you
+do not need to configure it explicitly.
+
+The `add_cloud_metadata` processor enriches each event with instance metadata
+from the machine's hosting provider. At startup the processor queries a list of
+hosting providers and caches the instance metadata.
+
+The following cloud providers are supported:
+
+* Amazon Web Services (AWS)
+* Digital Ocean
+* Google Compute Engine (GCE)
+* https://www.qcloud.com/?lang=en[Tencent Cloud] (QCloud)
+* Alibaba Cloud (ECS)
+* Huawei Cloud (ECS)
+* Azure Virtual Machine
+* Openstack Nova
+
+The Alibaba Cloud and Tencent cloud providers are disabled by default, because
+they require to access a remote host. Use the `providers` setting to select a
+list of default providers to query.
+
+[discrete]
+== Example
+
+This configuration enables the processor:
+
+[source,yaml]
+----
+  - add_cloud_metadata: ~
+----
+
+
+The metadata that is added to events varies by hosting provider. For examples,
+refer to <<provider-specific-examples>>.
+
+[discrete]
+== Configuration settings
+
+[options="header"]
+|===
+| Name | Required | Default | Description
+
+| `timeout`
+| No
+| `3s`
+| Maximum amount of time to wait for a successful response when detecting the hosting provider. If a timeout occurs, no instance metadata is added to the events. This makes it possible to enable this processor for all your deployments (in the cloud or on-premise).
+
+| `providers`
+| No
+|
+a| List of cloud provider names to use. If `providers` is not configured,
+all providers that do not access a remote endpoint are enabled by default.
+
+The list of supported cloud provider names includes:
+
+* `alibaba` or `ecs` for the Alibaba Cloud provider (disabled by default).
+* `azure` for Azure Virtual Machine (enabled by default).
+* `digitalocean` for Digital Ocean (enabled by default).
+* `aws` or `ec2` for Amazon Web Services (enabled by default).
+* `gcp` for Google Copmute Enging (enabled by default).
+* `openstack` or `nova` for Openstack Nova (enabled by default).
+* `openstack-ssl` or `nova-ssl` for Openstack Nova when SSL metadata APIs are enabled (enabled by default).
+* `tencent` or `qcloud` for Tencent Cloud (disabled by default).
+* `huawei` for Huawei Cloud (enabled by default).
+
+| `overwrite`
+| No
+| `false`
+| Whether to overwrite existing cloud fields. If `true`, the processor
+overwrites existing `cloud.*` fields.
+
+|===
+
+The `add_cloud_metadata` processor supports SSL options to configure the http
+client used to query cloud metadata.
+
+//TODO: Add documentation about SSL options. Do we want to pull the descriptions
```

Comment:
Need input from reviewers on this question. Do we need to document the SSL options under processors?